storage/netstore.go

```diff
@@ -18,318 +18,289 @@ package storage
 
 import (
 	"context"
-	"encoding/hex"
+	"errors"
 	"fmt"
 	"sync"
-	"sync/atomic"
 	"time"
 
-	"github.com/ethereum/go-ethereum/p2p/enode"
 	"github.com/ethersphere/swarm/chunk"
 	"github.com/ethersphere/swarm/log"
+	"github.com/ethersphere/swarm/network/timeouts"
 	"github.com/ethersphere/swarm/spancontext"
-	"github.com/opentracing/opentracing-go"
+	lru "github.com/hashicorp/golang-lru"
+
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/p2p/enode"
 	olog "github.com/opentracing/opentracing-go/log"
 	"github.com/syndtr/goleveldb/leveldb"
-
-	lru "github.com/hashicorp/golang-lru"
+	"golang.org/x/sync/singleflight"
 )
 
-type (
-	NewNetFetcherFunc func(ctx context.Context, addr Address, peers *sync.Map) NetFetcher
+const (
+	// capacity for the fetchers LRU cache
+	fetchersCapacity = 500000
```

Comment:
The `fetchers` LRU cache is responsible for removing stale fetchers (outgoing requests for chunks that were never delivered. 500k fetchers is equal to 28MB, and in practice we should never reach that many active fetchers).